### PR TITLE
Regex measurement

### PR DESCRIPTION
**Pull request by bjeram**
### Required for all PRs:

- [x ] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [?] Has appropriate unit tests.

Address Issue #6145 InfluxDB Database/Measurement Settings Based on MQTT Topic
using a modified regex processor that can change the measurement name ( if result_key = "measurement" ). At the moment it woeks just for tags, but can be extended to fileds as well.

### CODE REVIEW

**Comment by merouaneagar** [comment]
Hello,
We need this functionality

```
  [[processors.regex.tags]]
    ## tag that we use to 'calculate' the name of the measurment
    key = "topic"
    pattern = 'sensors/(\w+)/(\w+)/(\w+)'
    replacement = "${2}"
    result_key = "measurement"
```

Regards

**Comment by sjwang90** [comment]
Looks like this PR is a duplicate fo #6830. Feel free to open PR back up if I'm incorrect.

**Comment by bjeram** [comment]
It is duplicate of #6830, but #6830 was closed (as it was created by mistake).
So I suggest to reopen this PR, and if possible merge it

File: plugins/processors/regex/README.md [hunk 1]
```diff
@@ -19,6 +19,17 @@ For tags transforms, if `append` is set to `true`, it will append the transforma
     ## Matches of the pattern will be replaced with this string.  Use ${1}
     ## notation to use the text of the first submatch.
     replacement = "${1}xx"
+    
+  ## here is an example how to change the name of the measurement
+  ## if measurement is given to reult_key then it is changed the name os the measurement (not a tag!!)
```
Comment (sjwang90) [inline plugins/processors/regex/README.md:24]: please fix the typos in the comments

**Review by sjwang90 — CHANGES_REQUESTED** [review]

**Comment by srebhan** [comment]
@bjeram I think this can be closed as #9561 covers your functionality. Furthermore, #9995 provides an elegant way to extract information from MQTT topics. Please reopen if you disagree!

**Pull request closed without merging**